sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggregationIterator.scala

```diff
@@ -157,38 +180,89 @@ abstract class AggregationIterator(
       inputAttributes: Seq[Attribute]): (InternalRow, InternalRow) => Unit = {
     val joinedRow = new JoinedRow
     if (expressions.nonEmpty) {
-      val mergeExpressions = functions.zip(expressions).flatMap {
-        case (ae: DeclarativeAggregate, expression) =>
-          expression.mode match {
+      val filterExpressions = expressions.map(_.filter)
+      var isFinalOrMerge = false
+      val mergeExpressions = functions.zipWithIndex.collect {
```

Comment:
Why did you change `functions.zip(expressions).flatMap` to `functions.zipWithIndex.collect`?

Reply:
Yes，I should to avoid this change